FLEDGE_browser_bidding_and_auction_API.md

```diff
@@ -32,52 +32,316 @@ fetch('https://www.example-ssp.com/auction', {
 
 Exposing size of the blob is a temporary leak that we hope to mitigate in the future:
 
 1.  As TEEs become more prevalent and adtechs gain experience operating and deploying them, we can reconsider whether the server that the browser sends the blob to could in fact be a trusted one operating in a TEE.
 
 1.  The size of the encrypted blob may shrink considerably, where padding may become a more effective privacy prevention (e.g. imagine fixed size blobs) and may introduce little overhead. [We recently announced](https://github.com/WICG/turtledove/issues/361#issuecomment-1430069343) changes to the interest group update requirements that should facilitate significantly fewer interest groups. Having many fewer interest groups can greatly reduce the size of the encrypted blob. These same changes allow for more information to be stored in the real-time trusted bidding signals server and simply indexed with a small identifier kept in the interest group, again shrinking the interest groups.
+
+# Appendix
+
+## Request Blob Format
+
+The binary blob returned as the `request` from `navigator.getInterestGroupAdAuctionData()` contains the information needed for the [ProtectedAudienceInput](https://github.com/privacysandbox/fledge-docs/blob/main/bidding_auction_services_api.md#protectedaudienceinput) in a `SelectAd` B&A call, encrypted using HPKE with an encryption header like that used in [OHTTP](https://www.ietf.org/archive/id/draft-thomson-http-oblivious-01.html). The encryption is done using public keys the browser fetches from a trusted endpoint that correspond with private keys only shared with B&A servers running in TEEs. The `request` is derived from the interest groups stored in the browser to be provided to [the B&A API](https://github.com/privacysandbox/fledge-docs/blob/main/bidding_auction_services_api.md). The request includes the contents of the interest groups, and information that the B&A server can pass unmodified in `browserSignals` for bidding like `joinCount`, `bidCount`, and `prevWins`. The blob is padded to either one of the 7 allowed sizes to reduce the cross-site identity leaked by its size. In the event the `request` cannot be created the resulting array will have zero length.
+
+Prior to encryption the `request` is encoded as [CBOR](https://www.rfc-editor.org/rfc/rfc8949.html) with the following schema (specified using [JSON Schema](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-01)):
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "/BuyerInput/AuctionBlob",
+  "type": "object",
+  "properties": {
+    "version": { "type": "number" },
+    "compression": {
+      "enum": ["gzip", "brotli", "none"],
+      "default": "gzip"
+    },
+    "publisher": { "type": "string" },
+    "generationId": {
+      "type": "string",
+      "format": "uuid"
+    },
+    "enableDebugReporting": {
+       "type": "boolean",
+       "default": "false"
+    },
+    "interestGroups": {
```

Comment:
```suggestion
    "debugReportingKey": {
       "type": "string",
    },
    "interestGroups": {
```

Reply:
As per https://github.com/privacysandbox/fledge-docs/blob/main/debugging_protected_audience_api_services.md#reasons-for-using-a-secret-debug-token

Reply:
We can do this later.